core/src/test/java/org/elasticsearch/search/aggregations/metrics/scripted/InternalScriptedMetricTests.java

```diff
@@ -56,7 +88,26 @@ protected InternalScriptedMetric createTestInstance(String name, List<PipelineAg
         if (hasReduceScript) {
             reduceScript = new Script(ScriptType.INLINE, MockScriptEngine.NAME, REDUCE_SCRIPT_NAME, params);
         }
-        return new InternalScriptedMetric(name, randomAlphaOfLength(5), reduceScript, pipelineAggregators, metaData);
+        Object randomValue = randomValue(valueTypes, 0);
+        return new InternalScriptedMetric(name, randomValue, reduceScript, pipelineAggregators, metaData);
+    }
+
+    @SuppressWarnings("unchecked")
+    private static Object randomValue(Supplier<Object>[] valueTypes, int level) {
```

Comment:
it is also very specialized given that before dance that creates the suppliers, maybe wise to leave it here for now.